play/src/front/Components/Modal/ObjectDetails.svelte

```diff
@@ -0,0 +1,245 @@
+            $mapExplorationObjectSelectedStore.setPointedToEditColor(0x000000);
+        if ($mapExplorationObjectSelectedStore instanceof AreaPreview)
+            $mapExplorationObjectSelectedStore.setStrokeStyle(2, 0x000000);
+        if (holdEntity instanceof Entity) holdEntity.setPointedToEditColor(0x000000);
+        if (holdEntity instanceof AreaPreview) holdEntity.setStrokeStyle(2, 0x000000);
+
+        cleanPropertyComponents();
+    });
+
+    function cleanPropertyComponents() {
+        iconProperties.set(new Map());
+    }
+
+    function initPropertyComponents() {
+        cleanPropertyComponents();
+        // Create the properties buttons for the selected object
+        let newIconProperties = new Map<string, AddPropertyButtonType>();
+        if ($mapExplorationObjectSelectedStore instanceof Entity) {
+            for (const value of $mapExplorationObjectSelectedStore.getProperties()) {
+                let addPropertyButton = getAddPropertyButton(value.type);
+                if (addPropertyButton) newIconProperties.set(value.id, addPropertyButton);
+            }
+        }
+
+        if ($mapExplorationObjectSelectedStore instanceof AreaPreview) {
+            for (const value of $mapExplorationObjectSelectedStore.getAreaData().properties.values()) {
+                let addPropertyButton = getAddPropertyButton(value.type);
+                if (addPropertyButton) newIconProperties.set(value.id, addPropertyButton);
+            }
+        }
+        iconProperties.set(newIconProperties);
+    }
+
+    function getAddPropertyButton(type: string) {
+        let addPropertyButton: AddPropertyButtonType | undefined;
+        switch (type) {
+            case "jitsiRoomProperty":
+                addPropertyButton = {
+                    property: "jitsiRoomProperty",
+                };
+                break;
+            case "openWebsite":
+                addPropertyButton = {
+                    property: "openWebsite",
+                };
+                break;
+            case "playAudio":
+                addPropertyButton = {
+                    property: "playAudio",
+                };
+                break;
+            case "speakerMegaphone":
+                addPropertyButton = {
+                    property: "speakerMegaphone",
+                };
+                break;
+            case "listenerMegaphone":
+                addPropertyButton = {
+                    property: "listenerMegaphone",
+                };
+                break;
+            case "exit":
+                addPropertyButton = {
+                    property: "exit",
+                };
+                break;
+            case "silent":
+                addPropertyButton = {
+                    property: "silent",
+                };
+                break;
+            case "focusable":
+                addPropertyButton = {
+                    property: "focusable",
+                };
+                break;
+        }
```

Comment:
Here you should put a default case with a cast to a type never. This way, when we add a new property, the compiler will fail until we reference the property in this switch statement.

```suggestion
                break;
            default:
                const _exhaustiveCheck : never = type;
        }
```